java/memory/src/main/java/org/apache/arrow/memory/AllocationManager.java

```diff
@@ -47,22 +43,17 @@
  * typical query. The
  * contention of acquiring a lock on AllocationManager should be very low.
  */
-public class AllocationManager {
+public abstract class AllocationManager {
```

Comment:
rather than making this abstract, let's change to a BufferAllocator configuration that includes a AllocationManager factory. That way we avoid breaking encapsulation and can also have other configuration options carried across allocators.

Reply:
Thanks! It's reasonable but AllocationManager's constructor requires an `BaseAllocator` instance as a parameter, as a result, passing a AllocationManager factory to BufferAllocator's constructor brings dependency cycle (chicken & egg). Should we try decouple `BaseAllocator` from AllocationManager's constructor?

And probably I am not able to correctly getting this:
> rather than making this abstract, ...

Does this mean we should keep the `public class AllocationManager`, and make it extend a base class? Like what's in my first version?

Reply:
Why not just have the factory accept all the required items? Is there a reason this won't work?

Reply:
It seems a lambda-like factory should work properly, though I forgot what stopped me when I try using a factory at first. I'll duble-check later :)

Reply:
@jacques-n - Just changed to a config/factory approach. Could you please take a look? Thanks a lot.